File: common/rikmuld/camping/entity/tileentity/TileEntityLight.java
package rikmuld.camping.entity.tileentity;

import java.util.ArrayList;

import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.util.AxisAlignedBB;
import rikmuld.camping.core.util.CampingInvUtil;

public class TileEntityLight extends TileEntityMain {

	int tick;
	
	@Override
	public void updateEntity()
	{
		if(!worldObj.isRemote)
		{
			tick++;
			if(tick>10)
			{
				boolean flag = true;
				
				ArrayList<EntityPlayer> players = (ArrayList<EntityPlayer>) worldObj.getEntitiesWithinAABB(EntityPlayer.class, AxisAlignedBB.getBoundingBox(xCoord-2, yCoord-2, zCoord-2, xCoord+2, yCoord+2, zCoord+2));
				for(EntityPlayer player:players)
				{
					if(CampingInvUtil.hasLantarn(player))flag = false;
				}
				if(flag)worldObj.setBlock(xCoord, yCoord, zCoord, 0);
				if(!flag)tick=0;
			}
		}
	}
}

File: scala/com/rikmuld/camping/features/inventory_camping/TileEntityLight.scala
package com.rikmuld.camping.features.inventory_camping

import com.rikmuld.corerm.tileentity.TileEntitySimple
import net.minecraft.util.ITickable

class TileEntityLight extends TileEntitySimple with ITickable {
  var tick: Int =
    0

  override def update(): Unit =
    if(!world.isRemote) {
      tick += 1

      if (tick > 10)
        world.setBlockToAir(pos)
    }
}